[Saravanan Anuja Harish] ip

Harish-coding opened the pull request:
# Conan
**_Conan_** is a task manager, who can hep you manage your day easily :v:.
> "The key is not spending time, but investing it." - Stephen R. Covey [(Source)](https://www.inc.com/dan-scalco/22-time-management-quotes-to-inspire-you-to-achieve-your-goals.html) </br>

Conan draws inspiration from [Project Duke](https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/projectDuke/index.html), that was designed to help learn Software Engineering Techniques.

## Features
+ Run _**Conan**_ on your command line.
+ _**Conan**_ can remember your tasks from previous runs and can recollect them if you want to. :wink:

## User guide

Once you run the program in your Command line, Conan will introduce himself and ask for your name. 
Once you are acquainted you could tell Conan the tasks you want to do.

#### Tasks
There are :four: kinds of tasks that Conan can store.
1. **Todo** - Keeps track of the task that do not have to be completed without sometime. </br> Enter : todo {Task} in the command line for such commands.
2. **Event** - Any event that you have to attend an certain date and time, Conan can keep track of that for you. :stopwatch:
  Enter : event {Task} /by YYYY-MM-DD HHMM
3. **Deadline** - Tasks that need to be completed within a time frame. :hourglass:
Enter : deadline {Task} /by YYYY-MM-DD HHMM
4. **Task** - any general tasks to keep track of.
Enter : {Task , more than 2 words}

**Note**: Time argument HHMM is not necessary to enter.

#### Operations
+ list : To view the list of tasks lined up.
+ mark {Task Number}: Mark that task as complete.
+ unmark {Task Number}: Mark that task as not complete.
+ delete {Task Number}: Delete that task from the list.
+ due-on {YYYY-MM-DD}: To view tasks that are due on this date.
+ due-before {YYYY-MM-DD}: To view tasks that are due before this date.
+ find {Keyword}: To find tasks containing that keyword.
+ bye: To close the program. Conan stores all your tasks so that you can visit them later.

#### Remembering Tasks
Conan has the ability to remember tasks from your previous runs. If you want to access them, then run the program under the same name you used last time. :smile_cat:
And then type **yes** if you want to continue from last time, otherwise press **no**.

#### Abilities

- [X] Remember Tasks from previous runs, of a single user.
- [ ] Be able to save data of different users on the same Pc.
***
#### For programmers
Open IDE and run `Main.main()`;
```java
public class Main {
    public static void main(String[] args) {
       // Code implementation
    }
}
```

Commit:
Add Gradle support

Harish-coding committed:
Level - 1 is completed

Harish-coding committed:
Level-1

Harish-coding committed:
read me for level 1 updated

Harish-coding committed:
Level - 2 done

Harish-coding committed:
Level -3

Harish-coding committed:
Level-4

Harish-coding committed:
Text-Ui-Testing completed

Harish-coding committed:
Level-5 completed

Harish-coding committed:
Level-6

Harish-coding committed:
Enum completed

Harish-coding committed:
updated to store the users previous requested.

Harish-coding committed:
Level 8 completed

added DateHandler class, enabling more inituitive features than would allow users to find tasks they have to do before a set date.

Harish-coding committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Conan.java
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	text-ui-test/EXPECTED.TXT

Conflicts resolved.
New features:
-> can continue tasks from previous runs
-> ability to find tasks before a due date

Harish-coding committed:
A-More OOP

Additional classes that individually handle commands were added.

Harish-coding committed:
Put individual files in organised packages.

Harish-coding committed:
Added j-unit testing for testing each class. CommandChecker, DateHandler, and Task were tested. DukeTest was used as a sample.

Harish-coding committed:
Code documented using javadocs.

Harish-coding committed:
Code tweeked to comply with coding standards.

Harish-coding committed:
New find feature added that would enable users to search the tasks.

Harish-coding committed:
Merge branch 'branch-Level-9'

joelwongjy commented on src/main/java/CommandSet/AddCommand.java:
```diff
@@ -0,0 +1,104 @@
+
+    /**
+     * adds the task to taskList.
+     * @param taskList the list of user tasks.
```
Should there be an empty line between the description and parameter section?

Harish-coding replied:
Thanks for pointing out! Just now fixed it to include a line here, and in other relevant places.😄

joelwongjy commented on src/main/java/CommandSet/ByeCommand.java:
```diff
@@ -0,0 +1,35 @@
+     * @param taskList the list of user tasks.
+     */
+    public static void bye(String username, Storage storage, TaskList taskList) {
+
```
Maybe the name of the method could be an action like executeBye.

Harish-coding replied:
Thanks for the suggestion that seems more descriptive! 👍

joelwongjy commented on src/main/java/CommandSet/DueCommand.java:
```diff
@@ -0,0 +1,52 @@
+     * @param taskList the list of user tasks.
+     */
+    public static void tasksDueOn(String date, TaskList taskList) {
+        date = date.substring(DUE_ON.length()).trim();
```
Should the name be a verb? like getTasksDueOn

Harish-coding replied:
Yup, corrected it to be a verb, now looks better! 👍

joelwongjy commented on src/main/java/CommandSet/AddCommand.java:
```diff
@@ -0,0 +1,104 @@
+
+/**
+ * <h1>AddCommand<h1/>
+ * <p>
```
I like how you used HTML tags for JavaDocs

Harish-coding replied:
Thanks for noticing! 😃

joelwongjy commented on src/main/java/Helper/DateHandler.java:
```diff
@@ -0,0 +1,190 @@
+    /**
+     * checks whether the date is before the given date.
+     * @param maxDate the value if the date limit
+     * @return true if the true; false otherwise
```
I believe you missed out periods at the end of the line here

Harish-coding replied:
Yup fixed it! 👍

joelwongjy reviewed (commented):
LGTM! Just some minor styling issues to fix 😃

Harish-coding committed:
improved version after PR review comments given by  @joelwongjy.

Harish-coding committed:
read me file updated

Harish-coding committed:
Merge branch 'add-gradle-support' of https://github.com/Harish-coding/ip into gradle-support

Harish-coding committed:
Merge branch 'gradle-support'

Harish-coding committed:
finished adding gradle support.

aceszhenwei commented on src/main/java/CommandSet/FindCommand.java:
```diff
@@ -0,0 +1,36 @@
+     * @param taskList the list of user tasks.
+     */
+    public static void findTasksContaining(String word, TaskList taskList) {
+        word = word.substring(FIND.length()).trim();
```
Perhaps, in order to prevent recycling variables, this `word` could be renamed to `processedWord`?

aceszhenwei commented on src/main/java/Helper/DateHandler.java:
```diff
@@ -0,0 +1,195 @@
+
+        // handle time checking
+        if (userDate.contains(SPACE)) {
+            String hr_mt = userDate.split(SPACE, LIMIT)[TIME_IDX].trim();
```
Perhaps the String `hr_mt` could be renamed to something clearer. This could be to ensure there is no ambiguity.

aceszhenwei commented on src/test/java/Helper/DateHandlerTest.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    @Test
+    public void isOnDate_date_returnTrueOrFalse() {
+        DateHandler case1 = new DateHandler("2022-05-19");
```
Try not to use numbers to distinguish the names, in this case `case1`.

aceszhenwei commented on src/main/java/Helper/Ui.java:
```diff
@@ -0,0 +1,161 @@
+     *
+     * @param previousUser the previous username.
+     */
+    public static void printFoundSimilarName(String previousUser) {
```
Nice! This is just one example that I wanted to point out. Good to see that many names for methods and variables do not just differentiate between themselves, but also explain what they do.

aceszhenwei reviewed (commented):
Great job! I believe that your code has very few coding naming errors. LGTM!

Harish-coding committed:
Added GUI

Harish-coding committed:
files for checkstyle is added

Harish-coding committed:
Merge branch 'branch-A-CheckStyle'

Harish-coding committed:
file updates accoring to javadocs.

Harish-coding committed:
Include assertions: add few assertions in the program

There are several assumptions made while running the program.

There are might come situations when these assumptions are not met, but
the program would still be running.

We need a way to alert, the programmer that the program is being run as
expected. Let's add assertion statements at various points in the code,
to make sure our assumptions are met.

This way we would know where we went wrong and what exactly was missing.

Harish-coding committed:
Unclear code.

Needs to be updated to improve readability.

Let's change the code to better comply with the coding standards and
improve readability.

Harish-coding committed:
Merge pull request #2 from Harish-coding/branch-A-Assertions

Include assertions: add few assertions in the program

Harish-coding committed:
Merge branch 'master' into branch-A-CodeQuality

Harish-coding committed:
Merge pull request #3 from Harish-coding/branch-A-CodeQuality

Unclear code.

Harish-coding committed:
GUI is incomplete as error message only say try again.

User will find it hard to know what exactly is wrong.

GUI is improved to respond with appropriate error messages for different
situations.

Ui class is updated as it deals with the printing of messages to the
user.

Harish-coding committed:
Users can add tasks freely, without being notified if they have any
other engagements that day.

If there are any clashes the user can be notified so that they are not
double-booked for different tasks at the same time.

Add a feature to warn the users about possible clashes, by modifying the
AddCommand class. New methods have to be added to TaskList, Ui, and
Tasks and its children classes.

Modifying AddCommand class lets us deal with the issue at the
appropriate level.

Harish-coding committed:
User could add similar tasks.

This means that users could keep adding the same tasks, and this means
that conan stores duplicates of the same task.

Let's Modify the code so that new tasks added are not duplicates. If
they are duplicates then issue a warning to the user, and ask them to
delete the previous task if they would like to add the new task.

Modifying AddCommand and Ui class would help us deal with the problem at
the appropriate level.

Harish-coding committed:
Conan is running correctly.

Finalize product features and removed dead code.

Harish-coding committed:
Added a screenshot of the user interface, for reference.

Harish-coding committed:
tempStore file removed as it is not needed.

Harish-coding committed:
Certain Conan responses are vague.

This makes it hard for users to understand or feel connected to Conan.

Let's make a few improvements in DueCommand, FindCommand, and
ListCommand class, to make Conan more lively.

This way more of the code remains intact and small changes are local to
the relevant methods only.

Harish-coding committed:
Update README.md

README.md is updated to include a more detailed user guide.

Harish-coding committed:
README.md is updated to include a more detailed user guide. Contains
FAQ, and a glossary section for quick reference.

Harish-coding committed:
Update README.md

Harish-coding committed:
Set theme jekyll-theme-midnight

Harish-coding committed:
no message

Harish-coding committed:
Set theme jekyll-theme-cayman

Harish-coding committed:
Update README.md

Harish-coding committed:
Merge commit 'bf5dfe7e6e450274bdfcc3452fcc93773784dd00'

# Conflicts:
#	docs/README.md
# resolved temporarily

Harish-coding committed:
Merge branch 'master' of https://github.com/Harish-coding/ip

# Conflicts:
#	docs/README.md
# temporarily

Harish-coding committed:
ReadMe.md code is not correctly reflected on the website.

Makes it hard for the readers to understand.

Update the code by removing unnecessary code.

This way they are not reflected in the website page.

Harish-coding committed:
</br> appears in parts of the website headers.

Makes it difficult of users to read.

Remove the unnecessary </br> symbols in README.md.

Harish-coding committed:
Blockquote text is not formatted correctly.

Displays different text format than given by Conan.

Let's replace </br> with <br>.

To format the blockquote without too much change

Harish-coding committed:
The website includes instances of </br>.

The text is not properly laid out in the website.

Let's replace </br> with <br>, and add <br> in appropriate places.

This way the text is formatted properly without disturbing too much of
the code.

Harish-coding committed:
In the website, the colon is too close to italicised words.

Does not make colon visibly  easily.

Add a whitespace between the italicised words and colon.

Harish-coding committed:
Update README.md

Harish-coding committed:
Update README.md

Harish-coding committed:
Update README.md
